sw/device/lib/dif/dif_clkmgr.h

```diff
@@ -0,0 +1,200 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_CLKMGR_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_CLKMGR_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/clkmgr/doc/">Clock manager</a> Device Interface
+ * Functions
+ */
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * A toggle state: enabled, or disabled.
+ *
+ * This enum may be used instead of a `bool` when describing an enabled/disabled
+ * state.
+ */
+typedef enum dif_clkmgr_toggle {
+  /*
+   * The "enabled" state.
+   */
+  kDifClkmgrToggleEnabled,
+  /**
+   * The "disabled" state.
+   */
+  kDifClkmgrToggleDisabled,
+} dif_clkmgr_toggle_t;
+
+/**
+ * Hardware instantiation parameters for clock manager.
+ *
+ * This struct describes information about the underlying hardware that is
+ * not determined until the hardware design is used as part of a top-level
+ * design.
+ */
+typedef struct dif_clkmgr_params {
+  /**
+   * The base address for the clock manager hardware registers.
+   */
+  mmio_region_t base_addr;
+} dif_clkmgr_params_t;
+
+/**
+ * A handle to clock manager.
```

Comment:
Might be worth going through and checking the grammar on the comments. (The template instantiator is good but not perfect.)